compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -1103,6 +1104,11 @@ void OMR::ValuePropagation::transformArrayCopyCall(TR::Node *node)
    if (!comp()->getOptions()->needWriteBarriers())
       needWriteBarrier = false;
 
+   if (TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
+      {
+      needReadBarrier = true;
+      }
```

Comment:
Couldn't this code be simplified to

```c++
needReadBarrier = TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads();
```

or even

```c++
bool needReadBarrier = TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads();
```

Reply:
Yes. I've changed it to your second suggestion.